### PR TITLE
DS AL Solution: Quicksort in-place

### PR DESCRIPTION
**Pull request by DS-AL-Problem-Solver**
@Indrashish Please review.

### CODE REVIEW

**Comment by Indrashish** [comment]
Review in progress

File: src/com/dsalproblems/dsalproblems/ArrayReverse.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 import java.util.*;
 
 /**
- * This class holds the logic of reversing a given array
+ * This class holds the logic of reversing a given array.
```
Comment (Indrashish) [inline src/com/dsalproblems/dsalproblems/ArrayReverse.java:34]: Following proper punctuation process is a very good way of coding

**Comment by Indrashish** [comment]
Test cases tried:

1) 
         Input:  7
         6 15 219 32 1 -9 0
Output: Success

2) 
         Input:  1
         6
Output: Success

3) 
         Input:

Output: Success

4) 
         Input:  1

Output: Success

5) 
         Input:  3
         13 2 a
Output: Success

6)
         Input:  5
         23 1 45 -99999 12
Output: Success

7)
         Input:  5001
         1 1 2 3 -9 -8 -5 12 33 44 12 q
Output: Success

8)
         Input:  5
         23 12 1
Output: Success

9)
         Input:  3
         2 12# 1
Output: Success

10)
         Input:  100
         33 28 66 70 65 97 75 25 40 14 83 30 55 88 56 82 71 68 5 95 17 21 80 51 45 8 47 74 92 42 23 27 76 49 13 99 7 67 78 39 43 52 61 87 98 12 84 41 64 16 60 32 4 86 90 57 11 37 6 22 72 10 100 79 85 81 58 59 35 3 18 1 91 26 36 38 89 77 93 44 94 20 31 48 63 46 50 9 24 53 96 69 62 15 54 73 2 29 34 19
Output: Success

11) 
        Input: -3
        1 2 3
Output: Success, however as commented earlier, you might want to include a separate condition check for negative array size

12) 
        Input: 5 12 32767 3
Output: Success (Test case for multiple numbers in the first line of the input where we are supposed to have just a single number)

13)
        Input: a
        1 2 3
Output: Success

File: src/com/dsalproblems/main/Main.java [hunk 2]
```diff
@@ -52,4 +54,76 @@ private static void testBruteForceStringMatching() {
...
+            arraySize = Integer.parseInt(line);
+            System.out.println("Array size: " + arraySize);
+        } catch (NumberFormatException e) {
+            System.out.println("Array size should be a number.");
+            return;
+        }
+
+        if (arraySize < 1) {
+            System.out.println("Array size has to be at least 1.");
+        } else if (arraySize < 1 || arraySize > 5000) {
```
Comment (Indrashish) [inline src/com/dsalproblems/main/Main.java:79]: "else if(arraySize < 1 || arraySize > 5000)" 

Comment: In the previous if condition you have already checked if arraySize is less than 1 and returned appropriate error message. So, I think it is kind of redundant to do the same check again in the next if condition.

According to me, in order to be more specific, we can rephrase all the if conditions as follows:

if(arraySize < 0) {
        System.out.println("Array size cannot be negative");
}
else if( 0 < arraySize < 1) {
        System.out.println("Array size has to be at least 1");
}

else if(arraySize > 5000) { 
        System.out.println("Array size out of range");
}
else {
    ......
    ......
    ......  
}

In this way we can handle the test cases distinctively.

**Review by Indrashish — COMMENTED** [review]
#Accepted

**Comment by Indrashish** [comment]
#Accepted

File: src/com/dsalproblems/main/Main.java (hunk 2 above)
Comment (DS-AL-Problem-Solver) [reply src/com/dsalproblems/main/Main.java:79]: Good point!

However, I guess the following cannot happen.

```
else if( 0 < arraySize < 1) {
    System.out.println("Array size has to be at least 1");
}
```

Comment (DS-AL-Problem-Solver) [reply src/com/dsalproblems/main/Main.java:79]: According to me, a better condition check would have been:

```
if(arraySize < 0) {
    System.out.println("Array size cannot be negative");
} else if(arraySize < 1 || arraySize > 5000) {
    System.out.println("Array size out of range: " + "1 to 5000");
}
else {
    ......
}
```

**Pull request merged**